### ISSUE TITLE
improve the dep file tokenizer to remove OutOfMemory from the possible errors

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#6352]
https://github.com/ziglang/zig/blob/master/src-self-hosted/dep_tokenizer.zig

```zig
    pub fn init(allocator: *std.mem.Allocator, bytes: []const u8) Tokenizer {
```

```zig
    pub fn next(self: *Tokenizer) Error!?Token {
```

```zig
    const Error = error{
        OutOfMemory,
        InvalidInput,
    };
```

Tokenizers don't need allocators, so we can remove that complexity (and slowness / worse resource utilization) from this API.

It would also be nicer if the iterator didn't have a possible error. We could have a Token that represents invalid input, which if encountered the API user could then deal with.

This is a nice way to contribute to the self-hosted compiler since there are unit tests in this file :-)

**Comment by andrewrk** [comment]
Note to anyone taking this on, you'll want to wait until #6250 is landed into master before attempting this, I fixed a bunch of bitrot and renamed the file.

**Comment by andrewrk** [comment]
Landed in fe117d9961c3622fda5c359733d01de686509af0.
